### PR TITLE
EOS-19568-Kafka Sanity check added

### PR DESCRIPTION
**Pull request by tanujashinde0405**
Signed-off-by: Tanuja Shinde <tanuja.shinde@seagate.com>

### CODE REVIEW

File: srv/components/misc_pkgs/kafka/sanity_check.sls [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+Test kafka:
+  service.running:
+    - name: kafka
```
Comment (sradhanand-pati) [inline srv/components/misc_pkgs/kafka/sanity_check.sls:20]: check if the process is running or not.
refer to the older command we had used.

**Review by sradhanand-pati — DISMISSED** [review]

File: srv/components/misc_pkgs/kafka/sanity_check.sls [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+Test kafka:
```
Comment (ypise) [inline srv/components/misc_pkgs/kafka/sanity_check.sls:18]: This ensure the service is running, doesn't check if it's running.
What @83bhp meant was to use the tests:
- https://github.com/Seagate/cortx-prvsnr/commit/80c70e443d8fc35214d027466ae038a8056a0b23#diff-f2e1b47d101c4aa5f1493a7361ccb300dbbd4e4bfc7d9c510a219cd83a92e59cR22
- https://github.com/Seagate/cortx-prvsnr/commit/80c70e443d8fc35214d027466ae038a8056a0b23#diff-f2e1b47d101c4aa5f1493a7361ccb300dbbd4e4bfc7d9c510a219cd83a92e59cR41

Do update accordingly.

**Review by ypise — CHANGES_REQUESTED** [review]
Consider at least 5 iterations 1 sec apart to check if the service has come up before it reports failure.

**Review by ypise — APPROVED** [review]

File: srv/components/misc_pkgs/kafka/sanity_check.sls (hunk 2 above)
Comment (ypise) [reply srv/components/misc_pkgs/kafka/sanity_check.sls:21]: This command should be in loop as per my previous comment.

**Review by ypise — CHANGES_REQUESTED** [review]

File: srv/components/misc_pkgs/kafka/sanity_check.sls [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+{% for i in range(5) -%}
+Test Kafka-zookeeper:
+    cmd.run:
+        -name: test 1 -le $(ps ax | grep java | grep -i QuorumPeerMain | grep -v grep | awk '{print $1}' | wc -l)
+        - test.sleep:
```
Comment (sradhanand-pati) [inline srv/components/misc_pkgs/kafka/sanity_check.sls:22]: Can we have something like - 
https://github.com/Seagate/cortx-prvsnr/blob/a8a677d63646e4c626897f40ecbabb1106526317/srv/components/system/storage/multipath/config.sls#L92
```
Test kafka:
  cmd.run:
    - name: test 1 -le $(ps ax | grep ' kafka\.Kafka ' | grep java | grep -v grep | awk '{print $1}' | wc -l)
    - retry:
        attempts: 5
        until: True
        interval: 10
```

**Review by ypise — APPROVED** [review]

**Review by sradhanand-pati — APPROVED** [review]

**Pull request merged**